restapi/src/main/java/io/stargate/web/docsapi/service/DocsSchemaChecker.java

```diff
@@ -24,24 +24,30 @@ private void clearCacheOnSchemaChange(DocumentDB db) {
     }
   }
 
-  public void checkValidity(String keyspace, String table, DocumentDB db) {
+  public boolean checkValidity(String keyspace, String table, DocumentDB db, boolean throwOnFail) {
     ImmutableKeyspaceAndTable keyspaceAndTable =
         ImmutableKeyspaceAndTable.builder().keyspace(keyspace).table(table).build();
     clearCacheOnSchemaChange(db);
-    validatedDocCollections.computeIfAbsent(keyspaceAndTable, ks -> performValidityCheck(ks, db));
+    return validatedDocCollections.computeIfAbsent(
+        keyspaceAndTable, ks -> performValidityCheck(ks, db, throwOnFail));
   }
 
-  private boolean performValidityCheck(ImmutableKeyspaceAndTable keyspaceAndTable, DocumentDB db) {
+  private boolean performValidityCheck(
+      ImmutableKeyspaceAndTable keyspaceAndTable, DocumentDB db, boolean throwOnFail) {
```

Comment:
I'm not sure I understand the point of the `throwOnFail` flag (which in general boolean flags are a code smell). Why not have the caller catch that exception or add another method?

Reply:
Yea I removed this by just adding another method :) thanks